src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/InMemoryDirectoryInfo.cs

```diff
@@ -119,13 +119,11 @@ public override IEnumerable<FileSystemInfoBase> EnumerateFileSystemInfos()
 
         private bool IsRootDirectory(string rootDir, string filePath)
         {
-            if (!filePath.StartsWith(rootDir, StringComparison.Ordinal) ||
-                filePath.IndexOf(Path.DirectorySeparatorChar, rootDir.Length) != rootDir.Length)
-            {
-                return false;
-            }
+            int rootDirLength = rootDir.Length;
 
-            return true;
+            return filePath.StartsWith(rootDir, StringComparison.Ordinal) &&
+                (rootDir[rootDirLength - 1] == Path.DirectorySeparatorChar ||
```

Comment:
Minor nit - Apply [IDE0056](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0056):
```suggestion
                (rootDir[^1] == Path.DirectorySeparatorChar ||
```

Reply:
Can't use this language feature here since we target `netstandard2.0` and `net461` which doesn't support System.Index.